packages/schematics/angular/class/index.ts

```diff
@@ -34,7 +34,9 @@ export default function (options: ClassOptions): Rule {
 
     if (options.path === undefined) {
       const projectDirName = project.projectType === 'application' ? 'app' : 'lib';
-      options.path = `/${project.root}/src/${projectDirName}`;
+      options.path = project.sourceRoot
+        ? `/${project.sourceRoot}/${projectDirName}`
+        : `/${project.root}/${project.sourceRoot || 'src'}/${projectDirName}`;
```

Comment:
The code section inside this `if` block appears to be repeated in most schematics.  It probably should be pulled out and introduced as a utility function to simplify any changes/fixes in the future.

Reply:
Good idea... refactored.